packages/headless/src/features/document-suggestion/document-suggestion-slice.ts

```diff
@@ -0,0 +1,24 @@
+import {createReducer} from '@reduxjs/toolkit';
+import {fetchDocumentSuggestions} from './document-suggestion-actions';
+import {getDocumentSuggestionInitialState} from './document-suggestion-state';
+
+export const documentSuggestionReducer = createReducer(
+  getDocumentSuggestionInitialState(),
+
+  (builder) => {
+    builder
+      .addCase(fetchDocumentSuggestions.rejected, (state, action) => {
+        state.status.error = action.payload ?? null;
```

Comment:
Why is `?? null` needed? Wouldn't the rejected payload always contain an error?

Reply:
The payload for the rejected async thunk is actually defined as `CaseAssistAPIErrorStatusResponse | undefined` and this null check pattern can be found in other slice files because of this. For example "search-slice" and "product-listing-slice".